src/main/java/com/spider/solomon/order/CustomerOrderDao.java

```diff
@@ -12,6 +12,8 @@
 public class CustomerOrderDao {
 	@Resource
 	JdbcTemplate jdbcTemplate;
+	@Resource
+	OrderItemDao orderItemDao;
```

Comment:
What do you think of introducing a service class to aggregate this data, and having the controller call the service class instead of the DAO directly.

Reply:
That's a good question! Can you please tell us some of the pros and cons, and what your recommendation would be in this case?

Reply:
pros: 
- having a clear "controller -> service -> dao" organization allows a clear layer of responsibility of request/response definition, business logic, and data retrieval. 
- if you separate out aggregating the `OrderItem` retrieval, operations where you only need to retrieve order header information and not the items becomes more efficient because of less nested queries.

cons:
- The `CustomerOrder` object becomes a bit unintuitive because it may have an empty `items` field and it's a bit vague whether it should be filled automatically by the DAO layer.

Ultimately I would still want a service class and have the `dao` call moved there. I'd push for having separate "DAO models" and "response models" so that you can support immutability and not have the ambiguity of whether or not `items` is filled or not as described in the con list. Having objects pieced together in different methods makes for really hard troubleshooting and debugging.

Something I failed to mention as well was to create an index on the order item table for `orderId`.